samples/app-cache-meetings/nodejs/src/components/app-cache-tab.tsx

```diff
@@ -0,0 +1,119 @@
+// <copyright file="app-cache-tab.tsx" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// </copyright>
+
+import { useState } from "react";
+import React from "react";
+import * as microsoftTeams from "@microsoft/teams-js";
+
+/// </summary>
+/// function logitem
+/// </summary>
+function logItem(action: string, actionColor: string, message: string) {
+    const newItem =
+        "<span style='font-weight:bold;color:" +
+        actionColor +
+        "'>" +
+        action +
+        "</span> " +
+        message +
+        "</br>";
+    return newItem;
+}
+
+/// </summary>
+/// In beforeUnloadHandler using setItems and readyToUnload callback function
+/// </summary>
+const beforeUnloadHandler = (
+    setItems: React.Dispatch<React.SetStateAction<string[]>>,
+    readyToUnload: () => void) => {
+
+    let newItem = logItem("OnBeforeUnload", "purple", "Started");
+    setItems((Items) => [...Items, newItem]);
+
+    // dispose resources and cleanup
+    newItem = logItem("OnBeforeUnload", "purple", "Completed");
+    setItems((Items) => [...Items, newItem]);
+
+    console.log("sending readyToUnload to TEAMS");
+    readyToUnload();
+
+    return true;
+};
+
+/// </summary>
+/// loadHandler using setItems to set values
+/// </summary>
+const loadHandler = (
+    setItems: React.Dispatch<React.SetStateAction<string[]>>,
+    data: microsoftTeams.LoadContext) => {
+    logItem("OnLoad", "blue", "Started for " + data.entityId);
+
+    let newItem = logItem("OnLoad", "blue", "Completed for " + data.entityId);
+    setItems((Items) => [...Items, newItem]);
+
+    microsoftTeams.app.notifySuccess();
+};
+
+const AppCacheTab = () => {
+    const [items, setItems] = useState<string[]>([]);
+    const [title, setTitle] = useState("App Cache Sample");
+    const [initState] = useState(true);
+
+    React.useEffect(() => {
+        if (!initState) {
```

Comment:
what is the purpose of this initState? it is always true in your implementation.

 In kaela's original sample that was shared with you, notifySuccess was set after a timeout in a separate effect and this initState helped track that.
 
 This doesnt make sense in its current form since it is always true

Reply:
Removed un-used code.